src/pages/home/report/ReportActionCompose.js

```diff
@@ -323,6 +391,36 @@ class ReportActionCompose extends React.Component {
         this.setState({maxLines});
     }
 
+    /**
+     * Replace the code of emoji and update selection
+     * @param {Number} highlightedEmojiIndex
+     */
+    suggestedEmojisSelect(highlightedEmojiIndex) {
```

Comment:
I think a better function name for this would be `insertSelectedEmoji`:

```suggestion
    insertSelectedEmoji(highlightedEmojiIndex) {
```

Reply:
I personally found this a bit hard to understand, and I think it can be clarified with some local variable names:

```js
suggestedEmojisSelect(highlightedEmojiIndex) {
    const commentBeforeColon = this.state.value.slice(0, this.state.colonIndex);
    const emojiCode = this.state.suggestedEmojis[highlightedEmojiIndex].code;
    const commentAfterColonWithEmojiNameRemoved = this.state.value.slice(this.state.colonIndex).replace(/^:\w+($|\s)/, ' ');

    this.updateComment(`${commentBeforeColon}${emojiCode}${commentAfterColonWithEmojiNameRemoved}`);

    this.setState(prevState => ({
        selection: {
            start: prevState.colonIndex + 2,
            end: prevState.colonIndex + 2,
        },
    }));
}
```

Then beyond that I have two more comments on this function:

1. Should we reset the `colonIndex` at the end of the function along with the `selection`, since we'll have removed the `:emoji:` and replaced it with an emoji code?
1. `this.updateComment` takes a second parameter which does not have a default. I think this is an existing mistake and the second parameter of `this.updateComment` should have a default value of `true`.

Reply:
`colonIndex` is dynamically updated with each occurrence of the `onSelectionChange` event, ensuring that it remains up-to-date and relevant before being utilized.